src/components/forms/FormSearchField.js

```diff
@@ -1,43 +1,67 @@
 /**
- * @author psarando
+ * @author psarando sriram
  */
 import React from "react";
 
-import Autocomplete from "../../components/autocomplete/Autocomplete";
+import Autocomplete from "@material-ui/lab/Autocomplete";
 
 import getFormError from "./getFormError";
 
 import FormControl from "@material-ui/core/FormControl";
+import TextField from "@material-ui/core/TextField";
 import FormHelperText from "@material-ui/core/FormHelperText";
 
 const FormSearchField = ({
     field: { value, onBlur, onChange, ...field },
     label,
     helperText,
     required,
+    options,
+    size = "small",
+    renderCustomOption,
+    handleSearch,
     form: { touched, errors, setFieldTouched, setFieldValue },
     ...props
 }) => {
     const errorMsg = getFormError(field.name, touched, errors);
-    const onOptionSelected = (option) => {
+    const [searchValue, setSearchValue] = React.useState(value);
+
+    React.useEffect(() => {
+        const val = {};
+        val[props.labelKey] = value;
+        setSearchValue(val);
+    }, [value, setSearchValue, props.labelKey]);
+
+    const onOptionSelected = (event, option, reason) => {
         setFieldValue(field.name, option ? option[props.valueKey] : "");
     };
-    const onSearchBlur = (event) => {
-        setFieldTouched(field.name, true);
-    };
-
```

Comment:
Not sure how to do this in material-ui yet. There is `blurOnSelect`. May be use that ?

Reply:
I was testing this like you suggested by setting `required` to true but that didn't show any error when the field was left blank. But imo, it doesn't make sense to have a search field required. We should use `FormSelectField` for those 🤔

Reply:
This is just like any other text field, except it can search thousands of ontology terms the user can choose from, so they don't make a spelling mistake or typo with the term (Ramona also suggested that in the future the IRI could be saved in the metadata as well). The ontology terms in the API can also get updated from time to time. I don't think we can expect anyone to create a select field with thousands of options in our metadata templates.

The `onBlur` callback is still being removed from `field` in the destructuring at the top of this component, but that callback is how this field sets its `touched` to `true`, and the error message will only display if the field is touched.

Also `blurOnSelect` should probably be set, as you suggested earlier.

If this still does not allow the `TextField` to display the `Required` error message, then maybe the `{...field}` and `{...props}` need to be spread into the `TextField` instead (or maybe it should be a `FormTextField` which already has the logic of checking `getFormError` for its `helperText`, so we wouldn't have to do that here 🤔 ).

Reply:
No dice 🎲 . I have pushed all the changes I got. Further suggestions welcome.

Reply:
I got it to work in Storybook by adding the `required={required}` prop to the `TestField`, though now I'm seeing weird behavior:

If I clear the value, the field is marked as an error with the `Required` message when the value is cleared (as expected), but then selecting an option from the search results doesn't clear the validation error at first. I added console logging to the validation function that shows the new value is set, but then immediately after an empty value is set, even though the selection value still displays in the field 😕 
Selecting another option then clears the validation error, but again the validation function logging shows that the new value is set, followed immediately by the old value:
```
SearchField handleSearch = index.js:114
values.test_field <empty string> index.js:114
SearchField handleSearch = desert tree line ecotone index.js:114
values.test_field desert tree line ecotone index.js:114
values.test_field <empty string> index.js:114
SearchField handleSearch = desert-alpine tree line ecotone index.js:114
values.test_field desert-alpine tree line ecotone index.js:114
values.test_field desert tree line ecotone index.js:114
```
I'm not sure what's causing this yet.

Reply:
For some reason, removing `blurOnSelect` fixes this validation weirdness.